src/phasorpy/cluster.py

```diff
@@ -0,0 +1,263 @@
+      `Phasor-based image segmentation: machine learning clustering techniques
+      <https://doi.org/10.1364/BOE.422766>`_.
+      *Biomed Opt Express*, 12(6): 3410-3422 (2021).
+
+    Examples
+    --------
+    Recover the clusters from a synthetic distribution of phasor coordinates
+    with two clusters:
+
+    >>> real1, imag1 = numpy.random.multivariate_normal(
+    ...     [2, 3], [[0.3, 0.1], [0.1, 0.2]], 100
+    ... ).T
+    >>> real2, imag2 = numpy.random.multivariate_normal(
+    ...     [5, 6], [[0.2, -0.1], [-0.1, 0.3]], 100
+    ... ).T
+    >>> real = numpy.concatenate([real1, real2])
+    >>> imag = numpy.concatenate([imag1, imag2])
+    >>> (center_real, center_imag, radius_major, radius_minor, angle) = (
+    ...     phasor_cluster_gmm(real, imag, clusters=2)
+    ... )
+
+    """
+
+    real = numpy.asarray(real)
+    imag = numpy.asarray(imag)
+
+    if real.shape != imag.shape:
+        raise ValueError(f'{real.shape=} != {imag.shape=}')
+    if not isinstance(clusters, numbers.Integral) or clusters < 1:
+        raise ValueError(f"{clusters=} of type {type(clusters)}")
+    clusters = int(clusters)
+
+    skip_axis, _ = _parse_skip_axis(skip_axis, real.ndim)
+
+    if skip_axis:
+        real = numpy.mean(real, axis=skip_axis, keepdims=True)
+        imag = numpy.mean(imag, axis=skip_axis, keepdims=True)
+
+    coords = numpy.column_stack((real.ravel(), imag.ravel()))
+
+    valid_data = ~numpy.isnan(coords).any(axis=1)
+    coords = coords[valid_data]
+
+    if coords.shape[0] < clusters:
+        raise ValueError(
+            f'Not enough points ({coords.shape[0]}) for requested components ({clusters})'
+        )
+
+    kwargs.pop('n_components', None)
+
+    gmm = GaussianMixture(n_components=clusters, **kwargs)
+    gmm.fit(coords)
+
+    center_real = []
+    center_imag = []
+    radius_major = []
+    radius_minor = []
+    angle = []
+
+    for i in range(clusters):
+        center_real.append(float(gmm.means_[i, 0]))
+        center_imag.append(float(gmm.means_[i, 1]))
+
+        if gmm.covariance_type == 'full':
+            cov = gmm.covariances_[i]
+        elif gmm.covariance_type == 'tied':
+            cov = gmm.covariances_
+        elif gmm.covariance_type == 'diag':
+            cov = numpy.diag(gmm.covariances_[i])
+        else:  # 'spherical'
+            cov = numpy.eye(2) * gmm.covariances_[i]
+
+        eigenvalues, eigenvectors = numpy.linalg.eigh(cov[:2, :2])
+
+        idx = eigenvalues.argsort()[::-1]
+        eigenvalues = eigenvalues[idx]
+        eigenvectors = eigenvectors[:, idx]
+
+        major_vector = eigenvectors[:, 0]
+        current_angle = numpy.arctan2(major_vector[1], major_vector[0])
+
+        if current_angle < 0:
+            current_angle += numpy.pi
+
+        angle.append(float(current_angle))
+
+        radius_major.append(scaling * numpy.sqrt(eigenvalues[0]))
+        radius_minor.append(scaling * numpy.sqrt(eigenvalues[1]))
```

Comment:
These are all scalar values, right? Better use functions and constants from the `math` module instead of numpy.